lib/product.rb

```diff
@@ -0,0 +1,24 @@
+class Product
+  attr_reader :category,
+              :name,
+              :unit_price
+
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity
```

Comment:
Missing a hoarded attribute. Should start at false, #hoard should flip it to true, and #is_hoarded should report its contents/